api/database/migrations/2022_11_25_232602_would_accept_temporary_to_position_duration.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use Database\Helpers\ApiEnums;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Support\Facades\DB;
+
+class WouldAcceptTemporaryToPositionDuration extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->jsonb('position_duration')->nullable()->default(null);
+        });
+        Schema::table('applicant_filters', function (Blueprint $table) {
+            $table->jsonb('position_duration')->nullable()->default(null);
+        });
+
+        DB::statement(
+        <<<SQL
+            UPDATE users
+                SET position_duration =
+                    case would_accept_temporary
+                        when true then :wouldAccept::jsonb
+                        when false then :wouldNotAccept::jsonb
+                        when null then null
+                    end
+        SQL, [
+            'wouldAccept' => json_encode([ApiEnums::POSITION_DURATION_TEMPORARY, ApiEnums::POSITION_DURATION_PERMANENT]),
+            'wouldNotAccept' => json_encode([ApiEnums::POSITION_DURATION_PERMANENT]),
+        ]);
+
+        DB::statement(
+        <<<SQL
+            UPDATE applicant_filters
+                SET position_duration =
+                    case would_accept_temporary
+                        when true then :wouldAccept::jsonb
+                        when false then :wouldNotAccept::jsonb
+                        when null then null
+                    end
+        SQL, [
+            'wouldAccept' => json_encode([ApiEnums::POSITION_DURATION_TEMPORARY, ApiEnums::POSITION_DURATION_PERMANENT]),
+            'wouldNotAccept' => json_encode([ApiEnums::POSITION_DURATION_PERMANENT]),
+        ]);
+
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropColumn('would_accept_temporary');
+        });
+        Schema::table('applicant_filters', function (Blueprint $table) {
+            $table->dropColumn('would_accept_temporary');
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropColumn('position_duration');
+        });
+        Schema::table('applicant_filters', function (Blueprint $table) {
+            $table->dropColumn('position_duration');
+        });
+    }
```

Comment:
Since we are dropping the `would_accept_temporary` columns, they should be added back in the `down` method so we can step through rollbacks if necessary.

Reply:
ah snap, yeah!
done in  https://github.com/GCTC-NTGC/gc-digital-talent/pull/4956/commits/8ee7546b93c12ef53106a45829cdafcd9d73b908  and  https://github.com/GCTC-NTGC/gc-digital-talent/pull/4956/commits/1f98bb8b12f3ce2e918daf26e7449772fc57757f

couldn't get the reversal to work using `case`, error messages had me giving up, so I went with a bit of a different approach

Reply:
Thanks! I came back to check this was done 😛